### PR TITLE
Add a basic proptest for compactsize encodings.

### LINKED ISSUE #110: Proptest scaffold
Before doing #27 it would be convenient just to have a basic proptest skeleton so that the work of #27 or other property tests is just the work of writing the tests themselves.

### PR DESCRIPTION
**Pull request by hdevalence**
Closes #110.

### CODE REVIEW

**Comment by hdevalence** [comment]
This "finds" and fixes two bugs, for sort of cheating values of "find" (the first was a rustc warning, and the second I spotted while looking at the librustzcash code).

File: zebra-chain/src/serialization.rs [hunk 1]
```diff
@@ -273,3 +283,41 @@ pub trait ReadZcashExt: io::Read {
...
+    use proptest::prelude::*;
+
+    use std::io::Cursor;
+
+    proptest! {
+        // The tests below are cheap so we can run them a lot.
+        #![proptest_config(ProptestConfig::with_cases(100_000))]
+
+        #[test]
+        fn compactsize_write_then_read_round_trip(s in 0u64..0x2_0000u64) {
```
Comment (dconnolly) [inline zebra-chain/src/serialization.rs:299]: 👍

File: zebra-chain/src/serialization.rs [hunk 2]
```diff
@@ -273,3 +283,41 @@ pub trait ReadZcashExt: io::Read {
...
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use proptest::prelude::*;
+
+    use std::io::Cursor;
+
+    proptest! {
+        // The tests below are cheap so we can run them a lot.
+        #![proptest_config(ProptestConfig::with_cases(100_000))]
```
Comment (dconnolly) [inline zebra-chain/src/serialization.rs:296]: ,👍

File: zebra-chain/src/serialization.rs [hunk 3]
```diff
@@ -273,3 +283,41 @@ pub trait ReadZcashExt: io::Read {
...
+
+        #[test]
+        fn compactsize_write_then_read_round_trip(s in 0u64..0x2_0000u64) {
+            // Maximum encoding size of a compactsize is 9 bytes.
+            let mut buf = [0u8; 8+1];
```
Comment (dconnolly) [inline zebra-chain/src/serialization.rs:301]: 👍

File: zebra-chain/src/serialization.rs [hunk 4]
```diff
@@ -273,3 +283,41 @@ pub trait ReadZcashExt: io::Read {
...
+            Cursor::new(&mut buf[..]).write_compactsize(s).unwrap();
+            let expect_s = Cursor::new(&buf[..]).read_compactsize().unwrap();
+            prop_assert_eq!(s, expect_s);
+        }
+
+        #[test]
+        fn compactsize_read_then_write_round_trip(bytes in prop::array::uniform9(0u8..)) {
+            // Only do the test if the bytes were valid.
```
Comment (dconnolly) [inline zebra-chain/src/serialization.rs:309]: 👍

**Review by dconnolly — APPROVED** [review]
Lgtm! 🎉

**Pull request merged**